File: frontend/src/models/task.ts
// src/models/task.ts

export enum TaskStatus {
    PLANNED = "Planlagt",
    IN_PROGRESS = "I gang",
    COMPLETED = "Afsluttet",
    CANCELLED = "Annulleret"
}

export enum TaskPriority {
    LOW = "Lav",
    MEDIUM = "Medium",
    HIGH = "Høj",
    URGENT = "Akut"
}

export interface Task {
    id: string;
    title: string;
    description?: string;
    status: TaskStatus | string;
    priority: TaskPriority | string;
    installation_id?: string;
    created_date: string;
    due_date?: string;
    completed_date?: string;
    assigned_to?: string;
    estimated_hours?: number;
    actual_hours?: number;
    notes?: string;
}

export interface TaskCreate {
    title: string;
    description?: string;
    status: TaskStatus | string;
    priority: TaskPriority | string;
    installation_id?: string;
    due_date?: string;
    assigned_to?: string;
    estimated_hours?: number;
    notes?: string;
}

export interface TaskUpdate {
    title?: string;
    description?: string;
    status?: TaskStatus | string;
    priority?: TaskPriority | string;
    installation_id?: string;
    due_date?: string;
    completed_date?: string;
    assigned_to?: string;
    estimated_hours?: number;
    actual_hours?: number;
    notes?: string;
}

File: api/models/task.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class TaskBase(BaseModel):
    """Base model for Tasks."""
    title: str = Field(..., min_length=3, max_length=100)
    description: Optional[str] = None
    status: str = Field(..., description="Status for opgaven")
    priority: str = Field(..., description="Prioritet for opgaven")
    installation_id: Optional[str] = None
    due_date: Optional[datetime] = None
    assigned_to: Optional[str] = None
    estimated_hours: Optional[float] = None
    notes: Optional[str] = None

class TaskCreate(TaskBase):
    """Model for creating a new task."""
    id: Optional[str] = None  # Valgfrit ID, genereres ellers automatisk

class TaskUpdate(BaseModel):
    """Model for updating an existing task."""
    title: Optional[str] = Field(None, min_length=3, max_length=100)
    description: Optional[str] = None
    status: Optional[str] = None
    priority: Optional[str] = None
    installation_id: Optional[str] = None
    due_date: Optional[datetime] = None
    completed_date: Optional[datetime] = None
    assigned_to: Optional[str] = None
    estimated_hours: Optional[float] = None
    actual_hours: Optional[float] = None
    notes: Optional[str] = None

class TaskResponse(TaskBase):
    """Model for task response."""
    id: str
    created_date: datetime
    completed_date: Optional[datetime] = None
    actual_hours: Optional[float] = None

    class Config:
        from_attributes = True